tests/5.1/loop/test_loop_unroll.c

```diff
@@ -0,0 +1,52 @@
+//===--- test_loop_unroll.c -----------------------------------------------===//
+//
+// OpenMP API Version 5.1 August 2021 
+//
+// This test checks the behavior of the 5.1 unroll construct (2.11.9.2 in 5.1 spec). 
+// "The unroll construct fully or partially unrolls a loop", which is more of a code generation
+// compiler feature. 
+// This test specifically checks the unroll construct without any clause, in which the compiler decides
+// how the loop unrolls.
+// Referenced 5.1 Example Doc 7.2 
+////===--------------------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+#include <math.h>
+
+# define N 1024
+
+int errors; 
+
+int test_loop_unroll() {
+	int arr[N];
+	int sum = 0;
+	int parallel_sum = 0;
+
+	// sequential loop
+	for (int i = 0; i < N; i++) {
+		arr[i] = i;
+		sum += arr[i];
+	}
+
+	// omp unroll loop
+# pragma omp unroll
+	for (int i = 0; i < N; i++) {
+		parallel_sum += arr[i];
```

Comment:
parallel_sum is currently shared, cannot guarantee this will not cause a race condition.